network/src/commonMain/kotlin/com/wire/kalium/network/session/SessionManager.kt

```diff
@@ -63,3 +78,27 @@ fun HttpClientConfig<*>.installAuth(sessionManager: SessionManager) {
         }
     }
 }
+
+@OptIn(InternalAPI::class)
+private fun HttpClient.addWWWAuthenticateHeaderIfNeeded() {
+    receivePipeline.intercept(HttpReceivePipeline.Before) { response ->
+        if (response.status == HttpStatusCode.Unauthorized) {
+            val headers = buildHeaders {
+                appendAll(response.headers)
+                append(HttpHeaders.WWWAuthenticate, "Bearer")
+            }
+            proceedWith(
+                object : HttpResponse() {
```

Comment:
Annoying to have to override all the properties in order to update one value, but I guess it's no way around that.